### PR TITLE
Daniel Deng - Week 01 - Credit Card Validation

### PR DESCRIPTION
**Pull request by sinogermany**
The following functions have been implemented in this pull request:

| homework | exercise | function name | signature | example |
| --- | --- | --- | --- | --- |
| homework01 | exercise01 | `toDigits` | `Integer -> [Integer]` | `187` -> `[1, 8, 7]` |
| homework01 | exercise01 | `toDigitsRev` | `Integer -> [Integer]` | `187` -> `[7, 8, 1]` |
| homework01 | exercise02 | `doubleEveryOther` | `[Integer] -> [Integer]` | `[3, 7, 8, 1]` -> `6, 7, 16, 1` |
| homework01 | exercise03 | `sumDigits` | `[Integer] -> Integer` | `[1, 2, 11] -> 4` |
| homework01 | exercise04 | `validate` | `Integer -> Bool` | `-- credit card number valid or not` |

### CODE REVIEW

File: cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise02.hs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module CIS194.Homework01.Exercise02 (doubleEveryOther) where
+
+doubleEveryOtherFromLeft :: [Integer] -> [Integer]
+doubleEveryOtherFromLeft [] = []
+doubleEveryOtherFromLeft [element] = [element]
+doubleEveryOtherFromLeft (first:second:rest) = [first] ++ [second * 2] ++ doubleEveryOtherFromLeft rest
```
[the remark this answers is not in the document]
Comment (stevemao) [reply cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise02.hs]: Is there a linter to autofix whitespaces?

[the remark this answers is not in the document]
Comment (sinogermany) [reply cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise02.hs]: @shineli1984 Done. Thanks for pointing this out.
@stevemao I'm about to apply `hfmt` in the following exercises. Cheers.

File: cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise01.hs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module CIS194.Homework01.Exercise01 (toDigits, toDigitsRev) where
+
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num <= 0  = []
+  | otherwise = toDigits (num `div` 10) ++ [num `mod` 10]
```
[the remark this answers is not in the document]
Comment (cht8687) [reply cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise01.hs]: =s ? where is it ?

[the remark this answers is not in the document]
Comment (shineli1984) [reply cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise01.hs]: `=`s reads equal signs.
I was talking about the vertical alignment of the 2 equal signs on line 5 and 6

**Pull request merged**